### PR TITLE
Tidy up fonts and provide a font installation option

### PR DESCRIPTION
**Pull request by peterbrittain**
Hopefully this is the PR to resolve https://github.com/pwaller/pyfiglet/issues/59

### CODE REVIEW

File: pyfiglet/__init__.py [hunk 1]
```diff
@@ -46,6 +48,8 @@
 
 RESET_COLORS = b'\033[0m'
 
+SHARED_DIRECTORY = '%APPDATA%/pyfiglet/' if sys.platform == 'win32' else '/usr/local/share/pyfiglet/'
```
Comment (pwaller) [inline pyfiglet/__init__.py]: I'm expecting to find something which expands `%APPDATA%` into an actual path - or does windows interpret that during the `open()` function? Not a windows user so I can't test it.

File: pyfiglet/__init__.py [hunk 2]
```diff
@@ -169,6 +182,38 @@ def infoFont(cls, font, short=False):
...
+        # Copy the font definitions - unpacking any zip files as needed.
+        if os.path.splitext(file_name)[1].lower() == ".zip":
+            # Ignore any structure inside the ZIP file.
+            with zipfile.ZipFile(file_name) as zip_file:
+                for font in zip_file.namelist():
+                    font_file = os.path.basename(font)
+                    if not font_file:
+                        continue
+                    src = zip_file.open(font)
+                    dest = open(os.path.join(location, font_file), "wb")
```
Comment (pwaller) [inline pyfiglet/__init__.py]: This looks like we're missing  closes on these two opens? Can they be moved into a `with`?

File: pyfiglet/__init__.py (hunk 1 above)
Comment (peterbrittain) [reply pyfiglet/__init__.py]: Good spot!  I got a bit trigger happy on the PR.  meant to test that before submitting...  Looks like it needs an explicit conversion.  WIll change.

File: pyfiglet/__init__.py (hunk 2 above)
Comment (peterbrittain) [reply pyfiglet/__init__.py]: Doh!  That was just lazy programming...  Will fix.

**Comment by pwaller** [comment]
> I guess I can nest the with statements or drop python 2.6 support. Let me know which works for you.

The rationale I would take is "Let's not drop support for a version just for a want of a single extra statement and a bit of extra nesting". It would be frustrating to "almost have support", I think, given that the level of pain to have the support is minimal.

**Comment by peterbrittain** [comment]
If 2.6 was still supported, I'd totally agree.  However, it went EOL years ago (https://www.python.org/download/releases/2.6.9/), so I thought this might be a good time to remove support.  No worries - I'll push an update shortly.

**Comment by pwaller** [comment]
Thanks for that - do we want to include some links as to where you can find fonts?

**Comment by peterbrittain** [comment]
OK - I've updated the README.  As for testing?  I've run this code on:

* Windows 10, Python 3.4
  * Upload single font to a site-packages
  * Upload single font to shared directory (due to zipped egg install)
* Centos 7, Python 2.7
  * Upload single font and zip file to site-packages
  * Upload single font and zip file to shared directory

Good to go?

**Comment by pwaller** [comment]
I guess you might have missed my above comment, between your commit and your comment?

**Comment by peterbrittain** [comment]
Yup - that just appeared in my notifications...  Now done.

**Comment by pwaller** [comment]
I tried installing the jave fonts from http://www.jave.de/figlet/figletfonts40.zip, and doing `pyfiglet -L figletfonts40.zip`, but what I find is that I end up with zip-compressed fonts in my installation directory, which don't work.

```
pyfiglet/fonts$ file wavy.flf
wavy.flf: Zip archive data, at least v1.0 to extract
```

Do you have the same behaviour?

```pytb
$ pyfiglet -f wow a quick brown fox
Traceback (most recent call last):
  File "/home/pwaller/.local/src/github.com/pwaller/pyfiglet/pyfiglet/__init__.py", line 230, in loadFont
    raise FontError('%s is not a valid figlet font' % self.font)
pyfiglet.FontError: wow is not a valid figlet font

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/pwaller/.local/bin/pyfiglet", line 11, in <module>
    load_entry_point('pyfiglet', 'console_scripts', 'pyfiglet')()
  File "/home/pwaller/.local/src/github.com/pwaller/pyfiglet/pyfiglet/__init__.py", line 922, in main
    justify=opts.justify, width=opts.width,
  File "/home/pwaller/.local/src/github.com/pwaller/pyfiglet/pyfiglet/__init__.py", line 787, in __init__
    self.setFont()
  File "/home/pwaller/.local/src/github.com/pwaller/pyfiglet/pyfiglet/__init__.py", line 794, in setFont
    self.Font = FigletFont(font=self.font)
  File "/home/pwaller/.local/src/github.com/pwaller/pyfiglet/pyfiglet/__init__.py", line 120, in __init__
    self.loadFont()
  File "/home/pwaller/.local/src/github.com/pwaller/pyfiglet/pyfiglet/__init__.py", line 313, in loadFont
    raise FontError('problem parsing %s font: %s' % (self.font, e))
pyfiglet.FontError: problem parsing wow font: wow is not a valid figlet font
```

**Comment by peterbrittain** [comment]
Hmm...  I didn't use that zip file for my testing.  Looks like all the figlet fonts are actually also zip files and need decoding to plain text, despite their file extensions.

**Comment by pwaller** [comment]
Hmm. That is surprising! Maybe the solution to that is to have pyfiglet understand compressed fonts. I've filed that separately.

https://github.com/pwaller/pyfiglet/issues/63

**Pull request merged**